app/screens/edit_profile/edit_profile.tsx

```diff
@@ -0,0 +1,481 @@
+        defaultMessage: 'Last Name',
+    },
+    username: {
+        id: t('user.settings.general.username'),
+        defaultMessage: 'Username',
+    },
+    nickname: {
+        id: t('user.settings.general.nickname'),
+        defaultMessage: 'Nickname',
+    },
+    position: {
+        id: t('user.settings.general.position'),
+        defaultMessage: 'Position',
+    },
+    email: {
+        id: t('user.settings.general.email'),
+        defaultMessage: 'Email',
+    },
+};
+
+const EditProfile = ({
+    componentId,
+    currentUser,
+    isModal,
+    isTablet,
+    lockedFirstName,
+    lockedLastName,
+    lockedNickname,
+    lockedPosition,
+}: EditProfileProps) => {
+    const intl = useIntl();
+    const serverUrl = useServerUrl();
+    const theme = useTheme();
+    const keyboardAwareRef = useRef<KeyboardAwareScrollView>();
+    const firstNameRef = useRef<FloatingTextInputRef>(null);
+    const lastNameRef = useRef<FloatingTextInputRef>(null);
+    const usernameRef = useRef<FloatingTextInputRef>(null);
+    const emailRef = useRef<FloatingTextInputRef>(null);
+    const nicknameRef = useRef<FloatingTextInputRef>(null);
+    const positionRef = useRef<FloatingTextInputRef>(null);
+
+    const styles = getStyleSheet(theme);
+    const [userInfo, setUserInfo] = useState<UserInfo>({
+        email: currentUser.email,
+        firstName: currentUser.firstName,
+        lastName: currentUser.lastName,
+        nickname: currentUser.nickname,
+        position: currentUser.position,
+        username: currentUser.username,
+    });
+    const [canSave, setCanSave] = useState(false);
+    const [error, setError] = useState<ErrorText | undefined>();
+    const [updating, setUpdating] = useState(false);
+
+    const scrollViewRef = useRef<KeyboardAwareScrollView>();
+
+    const buttonText = intl.formatMessage({id: 'mobile.account.settings.save', defaultMessage: 'Save'});
+    const rightButton = useMemo(() => {
+        return isTablet ? null : {
+            id: 'update-profile',
+            enabled: false,
+            showAsAction: 'always' as const,
+            testID: 'edit_profile.save.button',
+            color: theme.sidebarHeaderTextColor,
+            text: buttonText,
+        };
+    }, [isTablet, theme.sidebarHeaderTextColor]);
+
+    const closeButtonId = 'close-edit-profile';
+    const service = currentUser.authService;
+
+    const leftButton = useMemo(() => {
+        return isTablet ? null : {
+            id: closeButtonId,
+            icon: CompassIcon.getImageSourceSync('close', 24, theme.centerChannelColor),
+            testID: closeButtonId,
+        };
+    }, [isTablet, theme.sidebarHeaderTextColor]);
+
+    useEffect(() => {
+        const unsubscribe = Navigation.events().registerComponentListener({
+            navigationButtonPressed: ({buttonId}: { buttonId: string }) => {
+                switch (buttonId) {
+                    case 'update-profile':
```

Comment:
Nit: This would better be in a constant.

Reply:
Done